src/Microsoft.IdentityModel.Tokens/Validation/Results/Details/ExceptionDetail.cs

```diff
@@ -57,17 +57,37 @@ public ExceptionDetail(
             };
         }
 
+        public ExceptionDetail(
+            MessageDetail messageDetail,
+            ValidationFailureType failureType,
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type exceptionType,
+            StackFrame stackFrame,
+            ExceptionDetail innerExceptionDetail)
+        {
+            InnerExceptionDetail = innerExceptionDetail;
+            MessageDetail = messageDetail;
+            _exceptionType = exceptionType;
+            FailureType = failureType;
+            StackFrames = new List<StackFrame>(4)
```

Comment:
Can you just do?

```suggestion
            StackFrames = [stackFrame];
```

If not, why default to a list of size 4?

Reply:
I was checking the allocations involved into using lists, and it appears that they start with a size of 0 and then double each time they reach their limit.
So doing `[stackFrame]`  would allocate it with a size of 1, then the next time a stack frame is added it would reallocate to 2, then 4, then 8, and so on.
I put the number 4 in a very precarious way while experimenting with it, and I believe we should consider what the minimum size of a stack trace would be for us to avoid re-allocating too many times as we add the different frames.

Reply:
That is fine.

It is interesting that you ended up at 4: https://github.com/dotnet/runtime/blob/66e45ef152002a2e1cbea939ded6643dfa48307c/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/List.cs#L23

It is the default if you don't specify an initial size.